packages/cli/src/ui/components/ThemeDialog.tsx

```diff
@@ -73,19 +89,75 @@ export function ThemeDialog({
     setFocusedSection('theme'); // Reset focus to theme section
   };
 
-  const [focusedSection, setFocusedSection] = useState<'theme' | 'scope'>(
-    'theme',
-  );
+  const handleCustomThemeSave = (customTheme: CustomTheme, scope: SettingScope) => {
+    if (onCustomThemeSave) {
+      onCustomThemeSave(customTheme, scope);
+    }
+    setShowCustomThemeEditor(false);
+    setSelectInputKey(Date.now()); // Refresh the theme list
+  };
+
+  const handleCustomThemeCancel = () => {
+    setShowCustomThemeEditor(false);
+  };
 
+  // Remove state and logic for 'create' focus section
+  // const [focusedSection, setFocusedSection] = useState<'theme' | 'create' | 'scope'>('theme');
+  const [focusedSection, setFocusedSection] = useState<'theme' | 'scope'>('theme');
+  const [selectedThemeIndex, setSelectedThemeIndex] = useState(initialThemeIndex >= 0 ? initialThemeIndex : 0);
+
+  // Add a handler for navigation
   useInput((input, key) => {
+    if (showCustomThemeEditor) return;
```

Comment:
remove all logic related to a custom theme editor from this pull request. expect most of the code in ThemeDialog should be reverted from this pr as the theme dialog should just behave as before with custom themes added to the list.